### PR TITLE
Maximum Nesting

### PR DESCRIPTION
**Pull request by dkrutskikh**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: maximum nesting metric

### CODE REVIEW

File: doc/metrics/maximum-nesting.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Maximum Nesting
+
+**Maximum Nesting** is the maximum nesting level of blocks / control structures that are present in a method. Methods with deep nesting level are often complex and tough to maintain.
```
Comment (incendial) [inline doc/metrics/maximum-nesting.md:3]: @dkrutskikh documentation mentions only methods and tests operate with only function. I'd suggest to add functions example to docs and methods example to tests then.

File: lib/src/metrics/nesting_level/nesting_level_visitor.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/source/line_info.dart';
+
+class NestingLevelVisitor extends RecursiveAstVisitor<Object> {
```
Comment (incendial) [inline lib/src/metrics/nesting_level/nesting_level_visitor.dart:5]: RecursiveAstVisitor isn't returning anything, so Object type could be replaced with void

File: test/reporters/code_climate_reporter_test.dart [hunk 3]
```diff
@@ -235,24 +237,54 @@ void main() {
         expect(report, isEmpty);
       });
 
-      test('without arguments', () {
+      test('with hight nesting level', () {
```
Comment (Anrock) [inline test/reporters/code_climate_reporter_test.dart:240]: hight => high

**Review by Anrock — APPROVED** [review]

File: test/reporters/html/utility_functions_test.dart [hunk 4]
```diff
@@ -1,6 +1,8 @@
+import 'package:dart_code_metrics/src/models/file_report.dart';
 @TestOn('vm')
```
Comment (incendial) [inline test/reporters/html/utility_functions_test.dart:2]: @dkrutskikh seems that the annotation should be first

File: doc/metrics/maximum-nesting.md (hunk 1 above)
Comment (dkrutskikh) [reply doc/metrics/maximum-nesting.md]: @incendial texts updated

File: lib/src/metrics/nesting_level/nesting_level_visitor.dart (hunk 2 above)
Comment (dkrutskikh) [reply lib/src/metrics/nesting_level/nesting_level_visitor.dart]: fixed

File: test/reporters/html/utility_functions_test.dart (hunk 4 above)
Comment (dkrutskikh) [reply test/reporters/html/utility_functions_test.dart:1]: fixed

**Review by incendial — APPROVED** [review]

**Pull request merged**